include/clock_control.h

```diff
@@ -36,10 +51,20 @@ typedef int (*clock_control_get)(struct device *dev,
 				 clock_control_subsys_t sys,
 				 u32_t *rate);
 
+typedef int (*clock_async_control)(struct device *dev,
+				   clock_control_subsys_t sys,
+				   struct clock_control_user *user,
+				   u32_t when);
```

Comment:
s/when/timeout would be better, and is a keyword used everywhere.

Reply:
maybe deadline as we want to indicate that we want it before that time. If it happens much earlier that is ok for requested module (it only impact power consumption)

Reply:
timeout has exactly that same meaning. See k_sem_take() for instance, or other function having a timeout as parameter.

Reply:
timeout it is.